fix: token prices and meaningful number formatting

lukeromanowicz opened the pull request:
<!--- Ensure the title of the Pull Request follows conventional commits syntax. If it resolves an issue, provide its ID in the scope section. -->

## Description
Resolves #481 

Fixes unreliable token prices and marketcaps  in token details page and account details -> tokens

## Demo
Examples:
account: ak_aeksJqKvh1HD4rSACRjSUGgVe4nEywq6C9VK6VYXwUMezePyH
tokens: ct_KeTvHnhU85vuuQMMZocaiYkPL9tkoavDRT3Jsy47LK2YqLHYb, ct_BwJcRRa7jTAvkpzc2D16tJzHMGCJurtJMUBtyyfGi2QjPuMVv
![image](https://github.com/aeternity/aescan/assets/46789227/a768866b-4220-4718-9c3f-5a6c1b76bbe4)
![image](https://github.com/aeternity/aescan/assets/46789227/62222425-6841-46d3-9a9f-afd294e8e342)


## Checklist:
<!--- Put an `x` in all the boxes that apply. -->
<!--- If your change requires a documentation PR, please link it appropriately -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] My change does not require a change to the documentation.

janmichek reviewed (changes requested):
Yeah, the prices seems to be much more relevant now 👍 
Thanks for the quick fix. Leaving some code suggestions

lukeromanowicz commented on src/utils/adapters.js:
```diff
@@ -369,11 +371,11 @@ export function adaptTokenDetails(token, totalSupply = null, price = null) {
 
   if (token && totalSupply) {
-    tokenDetails.totalSupply = Number(totalSupply / BigInt(10 ** token.decimals))
+    tokenDetails.totalSupply = (new BigNumber(totalSupply)).dividedBy(10 ** token.decimals).toNumber()
```
I'm afraid it should be done this way. The docs say we should use the operations on a newly constructed object, not the constructor itself. Take a look at examples: https://www.npmjs.com/package/bignumber.js/v/4.0.4

lukeromanowicz commented on src/stores/dex.js:
```diff
@@ -0,0 +1,57 @@
+import { defineStore } from 'pinia'
```
Well... my interpretation of that conversation is that we didn't agree on anything specific apart from dealing with this kind of duplicates at some point. I just asked for your opinion about moving this to a composable since this code is stateless but before I got the response I ended up implementing it in this store in order not to implement any new concepts. In my opinion, this is not a new concept since stores like `aesdk`, `blockchainStats`, `marketStats` or `websocket` also don't belong to a specific page and are widely used in many places.

janmichek replied:
I thought we concluded on leave it as it is, or moving it into format.js until we got more occurrences. So I am not sure what was the discussion for

lukeromanowicz replied:
The discussion was supposed to be about moving it to a composable but I decided on my own how to deal with it before you have answered.

janmichek replied:
Seems like the reused logic is now distributed on several places without a plan to improve it

lukeromanowicz replied:
I would say in this specific case it's actually stored in one place rather than multiple but I guess it's subjective. Anyway I created a followup task to address this and other cases of duplicated code for API calls https://github.com/aeternity/aescan/issues/485

lukeromanowicz commented:
Thank you for your review @janmichek I think I addressed all of the concerns. Let me know if I convinced you to keep it like it is or perhaps you insist on some changes.

michele-franchi commented:
The price is still not correct.

Current DEX price:
<img width="542" alt="image" src="https://github.com/aeternity/aescan/assets/11598532/598f3be2-d302-4b98-a738-dc18f850c5a8">

Current AE price:
<img width="355" alt="image" src="https://github.com/aeternity/aescan/assets/11598532/024428fe-6e34-4b9b-891b-4bed886ca7be">

So it should be:
<img width="348" alt="image" src="https://github.com/aeternity/aescan/assets/11598532/c5a65281-63ab-4a53-9aa6-0a53288f5724">

but aeScan shows:
<img width="1267" alt="image" src="https://github.com/aeternity/aescan/assets/11598532/0c9eb967-0581-4a97-9281-95e9a06ab78c">

lukeromanowicz commented:
@michele-franchi as we agreed on devchat I increased the significant digits for token values. I think it's good enough to push it to production, while the whole formatting issue requires some rethinking and refactoring in scope of https://github.com/aeternity/aescan/issues/484

janmichek commented on src/stores/tokenDetails.js:
```diff
@@ -47,15 +49,21 @@ export const useTokenDetailsStore = defineStore('tokenDetails', () => {
   function fetchTokenDetails(id) {
     tokenId.value = id
 
+    const tokenPromise = fetchToken()
```
Ok, thanks for explanation

michele-franchi reviewed (approved):
It looks good but as discussed the implementation can be improved, I have some ideas. Maybe I can pick up the followup since I originally implemented the `formatPrice` function.
